ui-tests/src/test/java/com/specmate/uitests/pagemodel/UITestUtil.java

```diff
@@ -1,20 +1,91 @@
 package com.specmate.uitests.pagemodel;
 
+import java.util.concurrent.TimeUnit;
+
 import org.openqa.selenium.By;
-import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.StaleElementReferenceException;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.support.ui.ExpectedConditions;
-import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.WebElement;
 
 public class UITestUtil {
+	/** 
+     * If the modal is displayed, we wait 1 second and then check again if the modal is still there
+     * The method checks for 10 seconds at intervals of 1 second (driver.findElement TimeOut 
+     * (defined by implictlyWait) (1 seconds) * counter (10)) if the modal disappeared
+     * 	*/
 	public static void waitForModalToDisappear(WebDriver driver) {
 		By modalLocator = By.id("loading-modal");
+		boolean modalDisplayed = true;
+    	int counter = 11;

```

Comment:
Wouldn't it be easier to use something like

new WebDriverWait(driver, 10)).until(ExpectedConditions.visibilityOfElementLocated(modalLocator);

instead of searching for the modal 10 times in a loop?

Reply:
Done